src/main/scala/sigmastate/eval/CompiletimeCosting.scala

```diff
@@ -50,9 +50,18 @@ trait CompiletimeCosting extends RuntimeCosting { IR: Evaluation =>
       case Terms.Apply(LongToByteArraySym, Seq(arg: Value[SLong.type]@unchecked)) =>
         eval(mkLongToByteArray(arg))
 
-      case sigmastate.Upcast(Constant(value, tpe), toTpe: SNumericType) =>
+      case Terms.Apply(ByteArrayToBigIntSym, Seq(arr: Value[SByteArray]@unchecked)) =>
+        eval(mkByteArrayToBigInt(arr))
+
+      case Terms.Apply(SigmaPropSym, Seq(bool: Value[SBoolean.type]@unchecked)) =>
+        eval(mkBoolToSigmaProp(bool))
+
+      case sigmastate.Upcast(Constant(value, _), toTpe: SNumericType) =>
```

Comment:
@greenhat I think we can do upcast and downcast not only constants but result of any expression of Numeric type.

Reply:
@greenhat no I mean it can be just any Value of Numeric type like `Upcast(Plus(IntConst(10), IntConst(20)), SLong)`. Why not?